file-manager/fileManager.cpp

```diff
@@ -51,9 +51,16 @@ bool upload(const std::string& filePath, mongocxx::collection& collection)
 	        kvp(FILE_DATA, data));
 
     // Insert the document into the collection.
-    collection.insert_one(doc.view());
-
-    std::cout << "Upload successful for: " << filePath << std::endl;
+    auto result = collection.insert_one(doc.view());
+    
+    if(result)
+    {
+        std::cout << "Upload successful for: " << filePath << std::endl;
+    }
+    else
+    {
+        std::cout << "Upload failed for: " << filePath << std::endl;
```

Comment:
Suggest reverting commit b28fafabafb588d56129e31936f6157a4bbc3367.

`insert_one` is expected to return an unset when unacknowledged write concern is used. That does not necessarily mean the operation failed. See 
http://mongocxx.org/api/current/classmongocxx_1_1collection.html#afa81175bce524eca00df540ede9487c1

> **Returns** The optional result of attempting to perform the insert. If the write concern is unacknowledged, the optional will be disengaged.

This example does not use unacknowledged write concern.

An exception is expected if the operation fails:

> **Exceptions** [mongocxx::bulk_write_exception](http://mongocxx.org/api/current/classmongocxx_1_1bulk__write__exception.html) if the operation fails.

Reply:
Ah good catch.